vendor/bat-native-ledger/BUILD.gn

```diff
@@ -19,20 +20,16 @@ config("external_config") {
     "include",
     "$target_gen_dir/include",
   ]
-  defines = [
-    "GEMINI_OAUTH_STAGING_URL=\"$gemini_oauth_staging_url\"",
-    "BITFLYER_STAGING_URL=\"$bitflyer_staging_url\"",
-  ]
 }
 
-config("internal_config") {
-  visibility = [
-    ":*",
-    "//brave/components/brave_rewards/test:*",
-    "//brave/test:*",
-    "//brave/vendor/bat-native-ledger/test:*",
-  ]
-  defines = [
+buildflag_header("buildflags") {
+  header = "buildflags.h"
+  flags = [
```

Comment:
I think we can put it into `src/bat/ledger` and include everywhere using `#include "bat/ledger/buildflags.h"` notion by adding:
```
header_dir = "vendor/bat-native-ledger/src/bat/ledger"
```